test/debug_info/procedure_argument.f90

```diff
@@ -0,0 +1,31 @@
+!RUN: %flang -g -S -emit-llvm %s -o - | FileCheck %s
+
+! check non debug instructions should not have debug location
```

Comment:
Could you clarify here that the symbols being bitcast below are static variables or literals? It would be nice to increase coverage to actual arguments that are local variables or dummy arguments.

Reply:
Updated the test case and commit message. The symbols being bitcast are function pointers.
The above explaination of the corresponding assembly code is not correct, which should be `example_func3` instead of `example_func1`. Updated now. The test case contains the arguments of constant value `2` and local variable `i`.

```
$ llvm-addr2line -p -f -e ./a.out 0x0000000000400988
example_func at /home/.../case.f90:0
$ llvm-addr2line -p -f -e ./a.out 0x00000000004009b8
example_func1 at /home/.../case.f90:0
```